File: Backend/routes/borrowed.py
from fastapi import APIRouter
from database.crud_borrowed import insert_borrowed_db, get_borrowed_db

borrowed = APIRouter()


@borrowed.post('/')
def insert_borrowed(id_book: str, id_user: str, borrow_date: str, returned_date: str):
    return insert_borrowed_db(id_book, id_user, borrow_date, returned_date)


@borrowed.get('/')
def get_book_is_borrowed(id_user: str, id_book: str = None, status: str = None):
    return get_borrowed_db(id_user, id_book, status)

File: webapp/src/action/API/borrowed.js

const URL_API = 'http://127.0.0.1:8000/'

export const getBorrowedByStatus = (idUser, status) => {
    return new Promise((resolve, reject) => {
        fetch(`${URL_API}borrowed/?id_user=${idUser}&status=${status}`)
            .then(response => response.json())
            .then(data => resolve(data))
            .catch(error => reject(error));
    });
}

export const getBookIsBorrowed = (idUser, idBook) => {
    return new Promise((resolve, reject) => {
        fetch(`${URL_API}borrowed/?id_user=${idUser}&id_book=${idBook}`)
            .then(response =>  response.json())
            .then(data => resolve(data))
            .catch(error => reject(error));
    });
}

export const insertBorrowed = (idBook, idUser, borrowDate, returnedDate) => {
    return new Promise((resolve, reject) => {
        fetch(`${URL_API}borrowed/?id_book=${idBook}&id_user=${idUser}&borrow_date=${borrowDate}&returned_date=${returnedDate}`, {
            method: 'POST',
            headers: {
              'Content-Type': 'application/json'
            },})
            .then(response =>  response.json())
            .then(data => resolve(data))
            .catch(error => reject(error));
    });
}


